src/crypto/CHIPCryptoPALmbedTLS.cpp

```diff
@@ -448,9 +448,25 @@ CHIP_ERROR add_entropy_source(entropy_source fn_source, void * p_source, size_t
     EntropyContext * const entropy_ctxt = get_entropy_context();
     VerifyOrReturnError(entropy_ctxt != nullptr, CHIP_ERROR_INTERNAL);
 
+#if (MBEDTLS_ALLOW_PRIVATE_ACCESS + 0) == 1
```

Comment:
Won't this run into -Wundef issues once we turn that on?

More generally, it seems like the right thing to do here is to keep track of whether we ever did add_entropy_source in this process, and if we did skip it, since the entropy sources are not removable...

Reply:
> Won't this run into -Wundef issues once we turn that on?

That might be true... I thought that it will be better to include this code *only* for the test case (I know it is a horrible practice in general), since in real world scenario no one will init/shutdown app in the same process (I think). But if you think that we can add simple "tracking" variable, it's OK for me.

Reply:
However, the problem that arrisess here, if we do not want to access "private" mbedTLS members, is that someone might add more than one entropy source with that function... So, it would be nice to track function address (and it's data) anyway. Making `add_entropy_source()` a one-off function does not seems nice. I will document that, but still...

Reply:
So the other option is: who is calling add_entropy_source and then presumably failing out if it fails?  Can _they_ track that they called it and not fail?

Reply:
I guess the Zephyr version of `PlatformManagerImpl::_InitChipStack`?

Reply:
Changed according to comment.